fud/fud/exec.py

```diff
@@ -109,28 +109,39 @@ def run_fud(args, config):
             try:
                 if ed._no_spinner:
                     sp.stop()
-                    result = ed.run(data, None)
+                    result = ed.run(data, args, sp=None)
                 else:
-                    result = ed.run(data, sp)
+                    result = ed.run(data, args, sp)
                 data = result
                 sp.end_stage()
             except errors.StepFailure as e:
                 sp.fail()
                 print(e)
                 exit(-1)
             durations.append(time.time() - begin)
-
         sp.stop()
 
-        if utils.is_debug():
-            utils.print_profiling_information("stages", path, durations)
+        # The case where overall stage times want to be printed.
+        if any(a == "all" for a in args.profile):
+            kwargs = {
+                "stage": "overall",
+                "phases": [ed for ed in path],
+                "durations": durations,
+            }
+            data = (
+                utils.profiling_csv(**kwargs)
+                if any(a == "csv" for a in args.profile)
+                else utils.profiling_information(**kwargs)
+            )
 
-        # output the data returned from the file step
         if args.output_file is not None:
-            if data.typ == SourceType.Directory:
+            if args.profile is not None:
+                with Path(args.output_file).open("wb") as f:
+                    f.write(str.encode(data))
+            elif data.typ == SourceType.Directory:
                 shutil.move(data.data.name, args.output_file)
             else:
                 with Path(args.output_file).open("wb") as f:
                     f.write(data.convert_to(SourceType.Bytes).data)
         elif data:
-            print(data.convert_to(SourceType.String).data)
+            print(data if args.profile else data.convert_to(SourceType.String).data)
```

Comment:
One minor annoyance here. Since these functions return `str`, we need to add an extra check. If the return type was `SourceType.String`, this wouldn't be necessary. Its unclear to me how to convert this from `str -> SourceType.String`.